### PR TITLE
Argument parsers messages and improvements

### PR DESCRIPTION
**Pull request by Xaxetrov**

### CODE REVIEW

File: src/ArgumentParser.cpp [hunk 1]
```diff
@@ -36,6 +36,7 @@ namespace cmake::language
 
     Error err;
     err.context = r.begin;
+    err.message = "Could not parse argument.";
```
Comment (FlorianChevassu) [inline src/ArgumentParser.cpp]: Maybe we could use the potential error messages returned by the "sub parsers" ?
We could produce something like:
```
Could not parse argument.
Note: Could be a BracketArgument but <BracketArgumentError>;
Note: Could be a QuotedArgumen tbut <QuotedArgumentError>;
Note: Could be a UnquotedArgument but <UnquotedArgumentError>
```

File: src/QuotedArgumentParser.cpp [hunk 2]
```diff
@@ -6,6 +6,14 @@
 
 namespace cmake::language
 {
+  tl::unexpected<Error> MissingEndingQuoteError(const Range& range)
```
Comment (FlorianChevassu) [inline src/QuotedArgumentParser.cpp]: Same comment as in #4 : I would call it `CreateMissingEndingQuoteError`.

File: src/UnquotedArgumentParser.cpp [hunk 3]
```diff
@@ -17,6 +17,8 @@ namespace cmake::language
     {
       Error err;
       err.context = r.begin;
+      err.message = "Could not parse unquoted argument.";
```
Comment (FlorianChevassu) [inline src/UnquotedArgumentParser.cpp]: We should use the error given by the UnquotedElement parser here

File: src/ArgumentParser.cpp (hunk 1 above)
Comment (FlorianChevassu) [reply src/ArgumentParser.cpp]: You still need to incorporate the intermediate error as children of the final error

**Comment by Xaxetrov** [comment]
Added a fluent builder to make error creation more streamlined and flexible.

**Review by FlorianChevassu — APPROVED** [review]
Great ! I like the ErrorBuilder !

**Pull request merged**